### PR TITLE
#trivial error serialization, request type check

### PR DESCRIPTION
**Pull request by bangtoven**
### _Summary_

- Fix missing error serialization
- Fix request arg type check
- Remove unnecessary `method` parsing on `serializeError`
- Syntax cleanup
- Prep for signer refactor

### _How did you test your changes?_

existing unit tests

### CODE REVIEW

File: packages/wallet-sdk/src/core/error/serialize.ts [hunk 1]
```diff
@@ -2,28 +2,21 @@
...
     shouldIncludeStack: true,
   });
 
   const docUrl = new URL('https://docs.cloud.coinbase.com/wallet-sdk/docs/errors');
   docUrl.searchParams.set('version', LIB_VERSION);
   docUrl.searchParams.set('code', serialized.code.toString());
-  const method = getMethod(serialized.data, requestOrMethod);
-  if (method) {
-    docUrl.searchParams.set('method', method);
-  }
```
Comment (bangtoven) [inline packages/wallet-sdk/src/core/error/serialize.ts:26]: this isn't used anyway.

File: packages/wallet-sdk/src/CoinbaseWalletProvider.ts [hunk 2]
```diff
@@ -36,17 +36,22 @@ export class CoinbaseWalletProvider extends EventEmitter implements ProviderInte
...
+      const { code } = error as { code?: number };
+      if (code === standardErrorCodes.provider.unauthorized) this.disconnect();
+      return Promise.reject(serializeError(error));
     }
   }
 
+  protected async handleRequest(request: RequestArguments) {
+    const category = determineMethodCategory(request.method) ?? 'fetch';
+    return this.handlers[category](request);
+  }
```
Comment (bangtoven) [inline packages/wallet-sdk/src/CoinbaseWalletProvider.ts:53]: prep for followup refactor to deprecate method category BS

File: packages/wallet-sdk/src/CoinbaseWalletProvider.test.ts [hunk 3]
```diff
@@ -32,7 +32,6 @@ describe('CoinbaseWalletProvider', () => {
     test('throws error when handling invalid request', async () => {
       const provider = createProvider();
       // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-      // @ts-expect-error // testing invalid request args
```
Comment (bangtoven) [inline packages/wallet-sdk/src/CoinbaseWalletProvider.test.ts:35]: we don't know what dapps would pass, and that's the whole point of us having `checkErrorForInvalidRequestArgs` assertion.
we actually should not expect error here.

File: packages/wallet-sdk/src/CoinbaseWalletProvider.test.ts [hunk 4]
```diff
@@ -72,7 +71,11 @@ describe('signer configuration', () => {
     (fetchSignerType as jest.Mock).mockRejectedValue(error);
 
     const provider = createProvider();
-    await expect(provider.request({ method: 'eth_requestAccounts' })).rejects.toThrow(error);
+    await expect(provider.request({ method: 'eth_requestAccounts' })).rejects.toMatchObject(
+      expect.objectContaining({
+        message: error.message,
+      })
+    );
```
Comment (bangtoven) [inline packages/wallet-sdk/src/CoinbaseWalletProvider.test.ts:78]: we should have thrown correctly serialized errors.
these tests were not correct.
fixed.

File: packages/wallet-sdk/src/CoinbaseWalletProvider.test.ts [hunk 5]
```diff
@@ -100,8 +107,9 @@ describe('signer configuration', () => {
 
   it('should throw error if signer is not initialized', async () => {
     const provider = createProvider();
-    await expect(provider.request({ method: 'personal_sign' })).rejects.toThrow(
-      `Must call 'eth_requestAccounts' before other methods`
-    );
+    await expect(provider.request({ method: 'personal_sign' })).rejects.toMatchObject({
+      code: standardErrorCodes.provider.unauthorized,
+      message: `Must call 'eth_requestAccounts' before other methods`,
+    });
```
Comment (bangtoven) [inline packages/wallet-sdk/src/CoinbaseWalletProvider.test.ts:113]: same as above.
the error should have been a proper error object containing message and a standard code.

**Review by fan-zhang-sv — APPROVED** [review]

**Pull request merged**